pokemongo_bot/cell_workers/egg_incubation_worker.py

```diff
@@ -0,0 +1,130 @@
+import json
+
+from pokemongo_bot.human_behaviour import sleep
+from pokemongo_bot import logger
+
+class EggIncubationWorker(object):
+    last_km_walked = 0
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.available_incubators = []
+        self.used_incubators = []
+        self.ready_eggs = []
+        self.incubating_eggs = []
+        self.km_walked = 0
+        self.remaining_km = []
+        EggIncubationWorker.last_km_walked
+
+    def work(self):
+        check_incubation = False
+        self._check_inventory() # get km, incubators, and incubating pkmn
+        self._calculate_remaining_km()
+        if len(self.remaining_km) and min(self.remaining_km)<=0:
+            logger.log('[#] Checking for hatched eggs...')
+            check_incubation = self._check_hatched_eggs()
+            sleep(10)
+            self.bot.latest_inventory = None
+            self._check_inventory() # incubators don't release til after check hatched eggs call
```

Comment:
The heartbeat current calls the hatched egg check. Not sure how that might affect your code.